### ISSUE TITLE
Logitech MX Master: Horizontal scrolling via scroll wheel doesn't work

### ISSUE THREAD

**Issue by wyldphyre** [microsoft/vscode#56964]
Issue Type: <b>Bug</b>

I've been using VS Code since shortly after it became a thing, and I think this has been an issue the whole time. It really needs to be fixed.

Horizontal scroll doesn't work with the horizontal scroll wheel on my Logitech MX Master (and presumably other similarly equipped mice). Typically whenever I try this the horizontal scroll moves slightly and then it refuses to work again, no matter how much I use the horizontal scroll wheel.

VS Code version: Code 1.26.1 (493869ee8e8a846b0855873886fc79d480d342de, 2018-08-16T18:38:57.434Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.54GB (3.29GB free)|
|Process Argv|C:\Users\craigr\AppData\Local\Programs\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (27)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
Bookmarks|ale|9.0.3
pascal|ale|6.3.0
spellright|ban|2.5.32
unique-lines|bib|1.0.0
bracket-pair-colorizer|Coe|1.0.59
markdown-table-prettify|dar|2.0.0
dart-code|Dar|2.17.1
vscode-markdownlint|Dav|0.20.0
xml|Dot|2.3.2
vscode-great-icons|emm|2.1.41
innosetup|idl|1.2.0
copy-markdown-as-html|jer|1.1.0
prettify-json|moh|0.0.3
python|ms-|2018.7.1
csharp|ms-|1.15.2
PowerShell|ms-|1.8.3
wordcount|ms-|0.1.0
preview-control-characters|nes|0.0.1
vscode-icons|rob|7.25.0
vscode-text-tables|Rom|0.1.5
code-settings-sync|Sha|3.0.0
vscode-fileutils|sle|2.10.3
vscode-autohotkey|sle|0.2.2
sort-lines|Tyr|1.7.0
vscode-todo-highlight|way|1.0.4
markdown-pdf|yza|1.2.0

</details>

**Comment by wyldphyre** [comment]
You've closed #38831 (which is essentially the same request as this one) as a duplicate of #13612, but #13612 doesn't seem to be related at all. It talks about laggy scrolling, not non-functional scrolling, and it doesn't mention horizontal scrolling at all.

**Comment by SanwariyaAmit** [comment]
Facing the same issue. Thank you for reopening it.

**Comment by bpasero** [comment]
@wyldphyre @SanwariyaAmit any difference if you configure `window.smoothScrollingWorkaround: true` ?

**Comment by SanwariyaAmit** [comment]
@bpasero Yes I have tried that, doesn't helps at all. 
I tried these two together as well.
`"window.smoothScrollingWorkaround": true
"window.titleBarStyle": "native"`

Two finger scrolling on touchpad works fine, but tilting the scroll wheel on mouse for horizontal scroll doesn't works.

**Comment by bpasero** [comment]
@SanwariyaAmit @wyldphyre any luck with this VSCode Exploration build that is using a much newer Electron (and thereby Chrome) version? https://az764295.vo.msecnd.net/exploration/a5cda44f872b9b55122a108b99a170e1ed5fa66e/VSCode-win32-x64-1.27.0-exploration.zip

**Comment by wyldphyre** [comment]
I tried the Exploration build and it's no better, with or without the smooth scrolling set. The horizontal scroll bar moves briefly, then stops, and then it won't respond to the scroll wheel again until the app is restarted.

Does it work for you?

**Comment by bpasero** [comment]
I do not own a Logitech MX Master, so I cannot say.

**Comment by bpasero** [comment]
@wyldphyre can you install [Electron Fiddle](https://github.com/electron/fiddle/releases/download/0.1/electron-fiddle-0.1.0-setup-win32-x64.exe) (another application that is using Electron) and run it and see if you can scroll in the editors of that application?

**Comment by wyldphyre** [comment]
I'm assuming it's not limited to my particular mouse. I just left work, but I'll test with Electron Fiddle tomorrow.

**Comment by SanwariyaAmit** [comment]
I use the Logitech M720 Triathlon. I tested out the Electron Fiddle and got the same results, there's a jerk (brief movement) on  trying to scroll but then it stops responding.

**Comment by wyldphyre** [comment]
My experience with Electron Fiddle was the same. The horizontal scrollbar moved briefly, once, and then no more.

Is VS Code known to work at all with horizontal scroll wheels? I've yet to hear of any confirmation that it works for someone.

**Comment by SanwariyaAmit** [comment]
This can be marked as a duplicate of #54024

**Comment by wolf99** [comment]
#8919 also relates to this

**Comment by mdnpascual** [comment]
Also has MX Master. I've only seen it once or twice working. Most of the time the horizontal scroll just doesn't work

**Comment by SanwariyaAmit** [comment]
Still facing the issue on Windows Build 1809 with Logitech Options 6.92.275

Previous Details:
Screencast of the Bug: https://www.screencast.com/t/3ot5VFKTBSEt
I'm using the Logitech M720 Triathlon
There's a jerk (brief movement) on trying to scroll but then it stops responding completely.

More Details:
Horizontal Scrolling on mouse doesn't works in VS Code stable, insider and exploration builds.
Also doesn't works in Electron Fiddle suggested by @bpasero
Works fine on the Monaco editor page: https://microsoft.github.io/monaco-editor/
window.smoothScrollingWorkaround: true doesn't helps at all.
Minimizing and restoring window fixes it, until resized.

**Comment by krupenja** [comment]
Just a note: I cannot repro this with MX Master in 1.29.1 on Ubuntu 18.10, scrolling using horizontal wheel works fine.

**Comment by wyldphyre** [comment]
I can confirm that horizontal scrolling now seems to work fine for me on Windows 10 using VS Code 1.29.1

**Comment by myfrom** [comment]
A question to people who report this is working: are you using User Setup version or the one that requires admin?

**Comment by wyldphyre** [comment]
I'm using the User Setup now. I was probably on Admin when I reported this.

**Comment by SanwariyaAmit** [comment]
Working for me as well. It was occurring on both types of setup.
The issue can be closed.

**Comment by SanwariyaAmit** [comment]
Spoken too soon, it breaks on resizing the window.
@wyldphyre Can you check if it continues to work after resizing the window.

**Comment by wyldphyre** [comment]
Looks like I spoke too soon. I just tried again with a random .cs file and horizontal scrolling is still busted for me. If I close VS Code and then open it again (so it opens straight into that file), it *sometimes* will scroll a bit, but typically it just does the usual "move a little bit, once, and then never again" behaviour.

It really did look fixed when I tried a file the other day, but right now it looks as broken as ever.

**Comment by Fanatic17** [comment]
I have a MX Master and I had a MX Anywhere 2 and they both suffer from this. I can scroll on chrome but not un VS code. If I scroll a bit on chrome and then come back maybe it works, but it's not a reliable fix. Also i remeber i did something with some registry key on my old computer to make this work on the Anywhere.

**Comment by FSM1** [comment]
also experiencing the same issues as reported above with an MX Master 2S

**Comment by takiden** [comment]
same issue, Windows 10, version 1803. VS Code 1.30.1

**Comment by intelliapps-io** [comment]
I have the same issue. I have an MX Master 2.

**Comment by SanwariyaAmit** [comment]
Can anyone from the US reachout to Logitech Support. They don't seems to have a support page to report bugs. The only contact is a US number +1 646-454-3200.

**Comment by wyldphyre** [comment]
For what it's worth, horizontal scrolling is currently working for me in VS Code 1.30.2.

**Comment by wolf99** [comment]
@wyldphyre to clarify, is that working horizontal scaling with an MX master mouse?
It is currently still not working at all for me in 1.30.2 - tested on two machines, one at the Windows 10 2018 autumn update (latest) and one at 1709

**Comment by wyldphyre** [comment]
@wolf99 Yes, it's horizontal scrolling with the first MX Master mouse. I was surprised when I noticed it was working!

While I hope it continues to work, who knows why it stated working so perhaps it will stop again. For now I hope it keeps doing what it's doing.

**Comment by verantiadesign** [comment]
Seems to be resolved, finally.

**Comment by thibka** [comment]
Nope, it's not resolved. I have the same issue with 1.31.0.
Horizontal scrolling is working in Chrome, but not in VS Code (Windows 10). It does work in VS Code on macOS though.

**Comment by philo23** [comment]
I've run into this issue as well with my Logitech MX Master (on Windows), but as mentioned earlier minimising and reopening vscode seems to make it start working again.

I'm beginning to suspect the actual issue is with the driver/Logitech options software, not vscode itself.

**Comment by takiden** [comment]
windows 10 Pro, build 1809, VSCode 1.31.1 
Logitech MX anywhere 2S works only after:
1. opening vscode 
2. minimizing it explicitly using the mouse by clicking the hyphen (super-key + down-arrow does not work)
3. restore the vscode window by clicking the icon in the taskbar

this procedure should be repeated every time you open vscode ... not nice at all X|

**Comment by dandrestor** [comment]
Same here as @takiden described.
On VSCode 1.31.1, Win 10 Enterprise 10.0.16299

**Comment by jamesw2037** [comment]
@takiden thanks for the little guide, I am experiencing the same issue with MX Master 2s and minimizing works. I wish there was a proper solution for this

VSCode 1.31.1 (user setup), Windows 10 Enterpriise 10.0.14393

**Comment by wyldphyre** [comment]
So it turns out despite my comment about horizontal scrolling working for me, that seems to not be the case, at least not all the time. At the time it seemed to be working but I had a case the other day where it wasn't. I guess it's just going to be one of those things.

**Comment by verantiadesign** [comment]
It seems to be working most of the time now, just sometimes when switching between windows, resizing the application window and switching virtual desktops it seems to stop working, it's not all the time (hard to reproduce at times). 

The work around for me is: Aim the cursor at the horizontal bar and use the sidescroller and then it works again.

**Comment by ravemir** [comment]
@verantiadesign 's workaround did not work for me, while the one mentioned by @takiden did.

**Comment by s-devaney** [comment]
@takiden workaround worked for me with an MX Ergo

**Comment by Crocmagnon** [comment]
I've noticed that I have to apply @takiden's workaround everytime I resize VS Code window, but it seems to work indeed.

**Comment by takiden** [comment]
@Crocmagnon if you are working on multiple external displays, then you'll have to apply this workaround when you move your VSCode window from one display to another...

I have to mention here, that I have tested the same mouse on OSX Mojave and Ubuntu 18.04 LTS and I didn't notice this faulty behaviour

**Comment by lukefan6** [comment]
It's also not working for Logitech M590 on Windows 10 Enterprise v.1803, vscode v.1.32.3
@takiden 's workaround works, though

**Comment by joel-felcana** [comment]
Same here. Working only after minimising and restoring after every time the window changes its size or position. If I change the window size or position, the horizontal scroll works for a single "jump" and then stops working. Two finger scrolling works, though

**Comment by spackmat** [comment]
I don't think this problem is directly Electron-specific. I have the exact same problem (scrolls some pixels horizontally and then refuses to do anything) also in Chromium-based browsers like Opera, but not, for example, in Excel or LibreOffice Calc or Firefox. So I assume there is an incompatibility between the horizontal scroll handling of Chromium and Logitech's implementation of this. So in my point of view, Logitech has to find a solution for that. Or are there mouses produces by other vendors with that problem?

**Comment by Crocmagnon** [comment]
After uninstalling Logitech Options, everything works fine for me on VS Code. This is not a solution though as it prevents from customizing the mouse behavior, but i thought it may be a useful insight.

**Comment by verantiadesign** [comment]
It used to work for a while, but since last update I cannot get it to work anymore. Seeing the last post, I start to think logitech options is the fault after all.

**Comment by SanwariyaAmit** [comment]
So I have been following this thread for quite a while.
I had shared the video on how to reproduce it and the workaround to fix it quite a while ago.

I had also asked if someone who's from USA can give Logitech Support (+1 646-454-3200) a call, since that's the only support option I found. Since the issue is inherent to Logitech Options (or maybe Chromium) and not VS Code.

**Comment by GunslingerBara** [comment]
This issue continues to occur with the latest VS Code and the just-released MX Master 3. Any updates on resolving this?

**Comment by tariva** [comment]
**_It works inthe current Versions with MXMaster S2!!!_**
My Versions:
VsCode:
1.38.0
3db7e09f3b61f915d03bbfa58e258d6eee843f35
x64
Election: 4.2.10
Chrome: 69.0.3497.128
NodeJS: 10.11.0
V8: 6.9.427.31-electron.9

LogitechOptions:
8.0.863
MX Master S2 Firmware:
012.000.00003
my settings on logitech Options
https://pasteboard.co/IwT8rN8.png
https://pasteboard.co/IwT8UkL.png

**Comment by bpasero** [comment]
Ok to close then?

**Comment by felipe1982** [comment]
Still not working for me

**Comment by bpasero** [comment]
@felipe1982 are you up for an experiment to see if this reproduces in Electron too? That would involve running https://github.com/atom/electron and opening a website in there to see if scrolling is any different?

**Comment by philo23** [comment]
This bug also happens in Spotify (horizontally scrolling recently played n the Home screen) and acts the exact same way (it works scrolls horizontally a little tiny bit once, then never again.) While Spotify isn’t actually Electron, it is using Chrome Embedded Framework from what I understand.

So I’m kinda convinced this isn’t a VSCode or even an Electron bug unfortunately, but some kind of issue with the way Logitech Options detects Chrome-based programs (except for Chrome itself) and then changes sending scroll events.

**Comment by felipe1982** [comment]
> @felipe1982 are you up for an experiment to see if this reproduces in Electron too? That would involve running https://github.com/atom/electron and opening a website in there to see if scrolling is any different?

@bpasero I downloaded the latest electron release from the atom/electron releases page, double clicked on electron.exe, and greeting an Electron welcome screen. I do not see a way to open a website, to then attempt to try horizontal scrolling.

**Comment by bpasero** [comment]
Given https://github.com/microsoft/vscode/issues/56964#issuecomment-540941104 I have to close this as a fix is not something VSCode can provide.

This issue seems to originate from an upstream component and is not something VSCode can fix by making a change to the code. As such, I will go ahead and close it. If you can identify the component that is causing the issue, please report it upstream:
* for UI, OS integration: https://github.com/electron/electron
* for Rendering: https://bugs.chromium.org/p/chromium/issues/list
* for file IO: https://github.com/nodejs/node

Thanks for understanding.
